include/private/common/libevent_reactor.hpp

```diff
@@ -195,62 +151,32 @@ class LibeventReactor : public Reactor, public NonCopyable, public NonMovable {
         }
     }
 
-    void pollin_once(socket_t fd, double timeo, Callback<Error> &&cb) override {
-        pollfd(fd, MK_POLLIN, timeo, [cb = std::move(cb)](Error err, short) {
-            cb(std::move(err));
-        });
+    static void pollfd_cb(short evflags, void *opaque) {
+        auto cbp = static_cast<mk::Callback<mk::Error, short> *>(opaque);
+        mk::Error err = mk::NoError();
+        assert((evflags & (~(EV_TIMEOUT | EV_READ | EV_WRITE))) == 0);
+        if ((evflags & EV_TIMEOUT) != 0) {
+            err = mk::TimeoutError();
+        }
+        // In case of exception here, the stack is going to unwind, tearing down
+        // the libevent loop and leaking forever `cbp` and the event once that
+        // was used to invoke this callback.
+        (*cbp)(std::move(err), evflags);
```

Comment:
Code from the C level callback moved here and removed code to convert from EV_READ and EV_WRITE to MK_POLLIN and MK_POLLOUT